addons/recruits/XEH_preInit.sqf

```diff
@@ -6,32 +6,46 @@ PREP_RECOMPILE_START;
 #include "XEH_PREP.hpp"
 PREP_RECOMPILE_END;
 
-if !(hasInterface) exitWith { ADDON = true; };
-
-// Register events
-["potato_safeStartOn", {
-    [GVAR(showMessage) || GVAR(showTags)] call FUNC(toggleRecruitInfo);
-}] call CBA_fnc_addEventHandler;
-
-["potato_safeStartOff", {
-    [false] call FUNC(toggleRecruitInfo);
-}] call CBA_fnc_addEventHandler;
-
-["ace_settingChanged", {
-    params ["_name", "_value"];
-
-    if (_name == QGVAR(showMessage) && missionNamespace getVariable [QEGVAR(safeStart,startTime_PV), -1] != -1) exitWith {
-        if (_value) then {
-            GVAR(oldSquad) = [];
-            [QGVAR(message)] call CFUNC(createRscTitle);
-        } else {
-            QGVAR(message) cutFadeOut 1;
-        };
-        [_value || GVAR(showTags)] call FUNC(toggleRecruitInfo);
-    };
-    if (_name == QGVAR(showTags) && missionNamespace getVariable [QEGVAR(safeStart,startTime_PV), -1] != -1) exitWith {
-        [GVAR(showMessage) || _value] call FUNC(toggleRecruitInfo);
-    };
-}] call CBA_fnc_addEventHandler;
+if (hasInterface) then {
+    [
+        QGVAR(showMessage),
+        "CHECKBOX",
+        ["Show Recuit Message", "Show the recruit message dialog."],
+        "POTATO UI",
+        true,
+        0,
+        {
+            if (missionNamespace getVariable [QEGVAR(safeStart,startTime_PV), -1] != -1) exitWith {
+                [_this] call FUNC(toggleRecruitInfo);
+            };
+        }
+    ] call cba_settings_fnc_init;
```

Comment:
settings should be added on all machines

Reply:
Do I have to worry about UI code running on server/HC?

Reply:
safer to run the setting init on the server as well

Reply:
Yeah, I agree they need to be set everywhere, more concerned about running display code on headless